### DISCUSSION TITLE
Dynamic variable for version name ... python interpreter notuseable.

### DISCUSSION THREAD

**Discussion by kif** [mesonbuild/meson-python#547] [Q&A] [answered]
Dear meson-python community,

I am using a dynamic variable to manage the version number in a PEP517 build environment. In the top-level `meson.build` file there is:
```
project('FreeSAS',
  'c', 'cython',
  license: 'MIT',
  meson_version: '>= 0.60',
  version: run_command(['version.py', '--wheel'],
                       check:true).stdout().strip(),
)
```
This builds nicely with `pip install .` but not when running `python -m build`, where the error message is:
```
The Meson build system
Version: 1.3.0
Source dir: /users/kieffer/workspace-400/freesas
Build dir: /users/kieffer/workspace-400/freesas/.mesonpy-4knfaa8f
Build type: native build

../meson.build:5:11: ERROR: Command `/users/kieffer/workspace-400/freesas/version.py --wheel` failed with status 1.

A full log can be found at /users/kieffer/workspace-400/freesas/.mesonpy-4knfaa8f/meson-logs/meson-log.txt
```
Unfortunately, the directory with the logs has already gone ... and I did not find a way to retrieve it.
I found that deactivating the build isolation was working better but different versions of meson were used: not really satisfactory. I also found that the `version.py` could not be executed in the isolated version because the python interpreter available there was not able to execute this simple script (!) 

So I wonder if you have any idea on how to debug this kind of cases, knowing that very similar projects (pyFAI, FabIO) are working with the very same structure.

Cheers,

Jerome

**Comment by dnicolodi** [comment, marked as answer]
I don't know what may be the difference between `pip` and `build` that causes this issue. The Meson log may have some more information. To access the build log you should specify a build directory explicitly: `python -m build -Cbuild-dir=build`

**Reply by kif** [reply]
Thanks a lot, this allows me to debug.
It confirms the python interpreter within the environment created by the `build` module is unable to parse/execute the python file. This explains why it works without the isolation and also the difference of behaviour with `pip`.

**Comment by dnicolodi** [comment]
> meson_version: '>= 0.60',

meson-python requires at least Meson 0.63.3, so this is not very useful.

**Reply by kif** [reply]
Thanks for the info.
